druid-shell/src/backend/x11/window.rs

```diff
@@ -1142,6 +1144,14 @@ impl Window {
         Ok(())
     }
 
+    pub fn handle_enter_notify(&self) {
+        self.with_handler(|h| h.got_focus());
```

Comment:
```rs
    /// Called when this window becomes the focused window.
    #[allow(unused_variables)]
    fn got_focus(&mut self) {}

    /// Called when this window stops being the focused window.
    #[allow(unused_variables)]
    fn lost_focus(&mut self) {}
```
Uhm... you are calling these when the mouse cursor enters/leaves the window, not when it is focused...?

Reply:
oh so these are mouse cursor enter/leave events :facepalm: . What are the keyboard focus events then?